File: src/main/java/com/temporal/agentic/activities/TemplateNodeActivities.java
package com.temporal.agentic.activities;

import com.temporal.agentic.models.*;
import java.util.*;

public interface TemplateNodeActivities {

    NodeExecutionResponse executeValidate(NodeExecutionRequest request);

    NodeExecutionResponse executeGovernance(NodeExecutionRequest request);

    NodeExecutionResponse executeRetrieve(NodeExecutionRequest request);

    NodeExecutionResponse executeTemplateMap(NodeExecutionRequest request);

    NodeExecutionResponse executeTransform(NodeExecutionRequest request);

    NodeExecutionResponse executeDecide(NodeExecutionRequest request);

    NodeExecutionResponse executeAction(NodeExecutionRequest request);

    NodeExecutionResponse executeFinalize(NodeExecutionRequest request);

    NodeExecutionResponse executePlanner(NodeExecutionRequest request);
}

File: src/main/kotlin/com/temporal/agentic/activities/TemplateNodeActivities.kt
package com.temporal.agentic.activities

import com.temporal.agentic.models.NodeExecutionRequest
import com.temporal.agentic.models.NodeExecutionResponse

interface TemplateNodeActivities {
    
    fun executeValidate(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeGovernance(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeRetrieve(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeTemplateMap(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeTransform(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeDecide(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeAction(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executeFinalize(request: NodeExecutionRequest): NodeExecutionResponse
    
    fun executePlanner(request: NodeExecutionRequest): NodeExecutionResponse
}
